src/linear_algebra/affine_space/barycentric_coords.lean

```diff
@@ -0,0 +1,96 @@
+/-
+Copyright (c) 2021 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import linear_algebra.affine_space.independent
+
+/-!
+# Barycentric coordinates
+
+Suppose `P` is an affine space modelled on the module `V` over the ring `k`, and `p : ι → P` is an
+affine-independent family of points spanning `P`. Given this data, each point `q : P` may be written
+uniquely as an affine combination: `q = w₀ p₀ + w₁ p₁ + ⋯` for some (finitely-supported) weights
+`wᵢ`. For each `i : ι`, we thus have an affine map `P →ᵃ[k] k`, namely `q ↦ wᵢ`. This family of
+maps is known as the family of barycentric coordinates. It is defined in this file.
+
+## The construction
+
+Fixing `i : ι`, and allowing `j : ι` to range over the values `j ≠ i`, we obtain a basis `bᵢ` of `V`
+defined by `bᵢ j = p j -ᵥ p i`. Let `fᵢ j : V →ₗ[k] k` be the corresponding dual basis and let
+`fᵢ = ∑ j, fᵢ j : V →ₗ[k] k` be the corresponding "sum of all coordinates" form. Then the `i`th
+barycentric coordinate of `q : P` is `1 - fᵢ (q -ᵥ p i)`.
+
+## Main definitions
```

Comment:
Could you add short descriptions to the list?

Reply:
Done.